cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -528,6 +540,29 @@ public Builder setDateTime(String fieldName, @Nullable LocalDateTime localDateTi
       return this;
     }
 
+    /**
+     * Sets the java.nio.ByteBuffer for a Fixed datatype field.
+     *
+     * @param fieldName String field name
+     * @param byteBuffer {@link ByteBuffer} value , is nullable
+     * @return {@link Builder} for the StructuredRecord
+     */
+    public Builder setFixed(String fieldName, @Nullable ByteBuffer byteBuffer) {
```

Comment:
Same as above. Can it be implemented in the normal `set` method logic?

Reply:
Because Fixed fields have to have an exact length, and fixed is a Logical type, I think it makes sense to have an explicit accessor to set these values (while also verifying the length of the byte array that is supplied)

Reply:
Same argument as above. If someone call the `set` method on a field with FIXED logical type, will it check the length? I think it should. This means essentially `set` and `setFixed` are doing the same job when the logical type is FIXED, implying there should be only one method, not two.

Reply:
The set method explicitly doesnot check the length.For each logical type(decimal,time,timestamp etc.) specific checks are performed in separate methods.So setFixed() specifically checks the length for Fixed type unlike set() method

Reply:
I think we should check the length. The set method should work for all types correctly (e.g. if a field has logical type of datatime, you can call the set method with a long, then later call the getDateTime method correctly).

Reply:
Set()  validates if Field is null
SetLogicalType  validates LogicalTypeSchema is null of given field for every logicaltype

Reply:
Removed setFixed and getFixed,implemented in set() method